pkg/api/norman/customization/cluster/validator.go

```diff
@@ -445,6 +462,34 @@ func validateAKSNodePools(spec *v32.ClusterSpec) error {
 	return nil
 }
 
+func validateAKSClusterName(client v3.ClusterInterface, spec *v32.ClusterSpec) error {
+	// validate cluster does not reference an AKS cluster that is already backed by a Rancher cluster
+	name := spec.AKSConfig.ClusterName
+	region := spec.AKSConfig.ResourceLocation
+	msgSuffix := fmt.Sprintf("in region [%s]", region)
+
+	// cluster client is being used instead of lister to avoid the use of an outdated cache
+	clusters, err := client.List(metav1.ListOptions{})
+	if err != nil {
+		return httperror.NewAPIError(httperror.ServerError, "failed to confirm clusterName is unique among Rancher AKS clusters "+msgSuffix)
+	}
+
+	for _, cluster := range clusters.Items {
+		if cluster.Spec.AKSConfig == nil {
+			continue
+		}
+		if name != cluster.Spec.AKSConfig.ClusterName {
+			continue
+		}
+		if region != "" && region != cluster.Spec.AKSConfig.ResourceLocation {
```

Comment:
What if the regions are both empty?

Reply:
Cluster with empty `region` can't be created, since it is hard requirement and validated.

Reply:
Is the `region != ""` not needed then? And if it were wouldn't the error that is returned be misleading?